### ISSUE TITLE
Allow HepMC output to distinguish jet shower particles from medium particles

### ISSUE THREAD

**Issue by jdmulligan** [JETSCAPE/JETSCAPE#63]
In the HepMC output, we should add a feature to encode whether a particle originates from the jet shower or from the medium. There is not currently an agreed upon standard in Rivet to do this, so here we would really be defining the convention. 

A good approach should be to add an extra "Attribute", which should be either a bool or an int. 
We need to be able to support the encoding in several different "modes" of running JETSCAPE, namely (i) writing out the full set of medium particles, and (ii) writing out only the jet shower + medium recoil/holes
So we want to include support for:
- jet particle -- from shower
- jet particle -- medium recoil or "hole"
- medium particle

One should also bear in mind that any Rivet routine will have to use this information appropriately, i.e. a single scheme of selecting certain attribute status will not be universal.

@kkauder @cnattras @bardelch

**Comment by kkauder** [comment]
Good idea. I think an int is much better. No reason to restrict ourselves to a binary choice. A string may also work, it's more descriptive than a number that needs an external table to interpret, but I don't know how well that works for rivet. Your final sentence is the crux, but once we know if rivet can make use of string attributed in a natural manner, maybe we should just go ahead and add the attribute, then let others follow our lead in this newly set standard.
